src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities/StackUtilities.cs

```diff
@@ -13,5 +18,39 @@ public static string GenerateStacksFilename(IEndpointInfo endpointInfo, bool pla
             string extension = plainText ? "txt" : "json";
             return FormattableString.Invariant($"{Utilities.GetFileNameTimeStampUtcNow()}_{endpointInfo.ProcessId}.stacks.{extension}");
         }
+
+        public static async Task CollectStacksAsync(TaskCompletionSource<object> startCompletionSource,
+            IEndpointInfo endpointInfo,
+            ProfilerChannel profilerChannel,
+            bool plainText,
+            Stream outputStream, CancellationToken token)
+        {
+            var settings = new EventStacksPipelineSettings
+            {
+                Duration = Timeout.InfiniteTimeSpan
+            };
+            await using var eventTracePipeline = new EventStacksPipeline(new DiagnosticsClient(endpointInfo.Endpoint), settings);
+
+            Task runPipelineTask = await eventTracePipeline.StartAsync(token);
+
+            //CONSIDER Should we set this before or after the profiler message has been sent.
```

Comment:
Before is fine, this is mostly to allow the event pipe session to have started before signaling the collection rule that it may resume the runtime of the target process when using a startup trigger.